src/librustc_metadata/decoder.rs

```diff
@@ -980,14 +980,7 @@ impl<'a, 'tcx> CrateMetadata {
     }
 
     fn get_attributes(&self, item: &Entry<'tcx>, sess: &Session) -> Vec<ast::Attribute> {
-        item.attributes
-            .decode((self, sess))
-            .map(|mut attr| {
-                // Need new unique IDs: old thread-local IDs won't map to new threads.
-                attr.id = attr::mk_attr_id();
```

Comment:
This is due to the related change which implement Decodable/Encodable on `AttrId` via `mk_attr_id` and `encode_nil`. I'm not sure if they're actually equivalent, though. It feels like they should be.

Reply:
Is this the only place where attributes are decoded?
Perhaps incremental could decode them in the old way as an index, or something like that?

Reply:
AFAICT, this is the only place where attributes are decoded, but I'm not sure how to check (cc @eddyb?)

I'm not sure what you mean by "incremental could decode them in the old way" -- is that referring to a place where it'd be good to check whether we're doing the right decoding?

Reply:
>AFAICT, this is the only place where attributes are decoded

Yeah, looks like that, at least from searching `.decode(` and `::decode(`.
I thought about maybe commenting out the impl and trying to build, but looks like that's not feasible because everything including attributes derives `RustcEncodable` and `RustcDecodable`.
(By incremental I meant some place in incremental compilation that decodes things from cache.)